lib/build-cloud.rb

```diff
@@ -219,17 +222,22 @@ def recursive_interpolate_config(h)
             h.map { |v| recursive_interpolate_config(v) }
         when String
 
-            while h =~ /%\{(.+?)\}/
-                var = $1
+            while /%\{(?<var>[^\|\}]*)(?:\|{2}(?<default>[^\}]*))?\}/ =~ h
+
+                exp = $&
                 val = ""
 
                 if @config.has_key?(var.to_sym)
```

Comment:
```
val = @config.fetch(var.to_sym, default)
raise "Attempt to interpolate with non-existant key '#{var}' with no default value set" if val.nil?
```
This should shortcut this `if` block. `default` will be set to `nil` by the regex on keys like `%{thing}`.